### PR TITLE
lsc mag

### PR DESCRIPTION
**Pull request by sys1874**

### CODE REVIEW

File: ogb_examples/lsc/MAG240M/dataset/base_dataset.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+            buffer.append(line_example)
+        else:
+            index = np.random.randint(0, len(buffer))
+            batch.append(buffer[index])
+            buffer[index] = line_example
+            if len(batch) == batch_size:
+                yield batch
+                batch = []
+
+#     np.random.shuffle(buffer)
```
Comment (Yelrose) [inline ogb_examples/lsc/MAG240M/dataset/base_dataset.py:84]: problem

File: ogb_examples/lsc/MAG240M/dataset/data_generator.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+        log.info(f'Done! [{time.perf_counter() - t:.2f}s]')
+
+    @property
+    def train_examples(self, ):
+        # Filters
+        trainer_id = int(os.getenv("PADDLE_TRAINER_ID", "0"))
+        trainer_num = int(os.getenv("PADDLE_TRAINERS_NUM", "1"))
+        count_line = 0
+
+        np.random.shuffle(self.train_idx)
```
Comment (Yelrose) [inline ogb_examples/lsc/MAG240M/dataset/data_generator.py:83]: Not Shuffle Here, seed is not the same

**Review by Yelrose — CHANGES_REQUESTED** [review]

File: ogb_examples/lsc/MAG240M/dataset/base_dataset.py (hunk 1 above)
Comment (sys1874) [reply ogb_examples/lsc/MAG240M/dataset/base_dataset.py]: fix

File: ogb_examples/lsc/MAG240M/dataset/data_generator.py (hunk 2 above)
Comment (sys1874) [reply ogb_examples/lsc/MAG240M/dataset/data_generator.py]: fix

**Review by Yelrose — APPROVED** [review]
LGTM

**Pull request merged**